core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -670,17 +671,17 @@ public void failEngine(String reason, @Nullable Exception failure) {
         if (failEngineLock.tryLock()) {
             store.incRef();
             try {
+                if (failedEngine.get() != null) {
+                    logger.debug((Supplier<?>) () -> new ParameterizedMessage("tried to fail engine but engine is already failed. ignoring. [{}]", reason), failure);
```

Comment:
Should we make this a `logger.warn` instead?  E.g. it could be that this new exception is the true root cause (e.g. disk full or something) whereas because of thread scheduling another (cascaded) exception (like IW's ACE) called `failEngine` first?